tracer/test/Datadog.Trace.Tests/Logging/TracerFlare/DebugLogReaderTests.cs

```diff
@@ -0,0 +1,154 @@
+﻿// <copyright file="DebugLogReaderTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.Logging.TracerFlare;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.Tests.Logging.TracerFlare;
+
+public class DebugLogReaderTests(ITestOutputHelper output)
+{
+    [Fact]
+    public void TryToCreateSentinelFile_CreateSentinelInExistingDirectory_Succeeds()
+    {
+        var directory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+        Directory.CreateDirectory(directory);
+
+        var id = Guid.NewGuid().ToString();
+        var result = DebugLogReader.TryToCreateSentinelFile(directory, id);
+        result.Should().BeTrue();
+    }
+
+    [Fact]
+    public void TryToCreateSentinelFile_CreateSecondSentinelInExistingDirectory_Succeeds()
+    {
+        var directory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+        Directory.CreateDirectory(directory);
+
+        var id1 = Guid.NewGuid().ToString();
+        var id2 = Guid.NewGuid().ToString();
+
+        var result1 = DebugLogReader.TryToCreateSentinelFile(directory, id1);
+        var result2 = DebugLogReader.TryToCreateSentinelFile(directory, id2);
+
+        result1.Should().BeTrue();
+        result2.Should().BeTrue();
+    }
+
+    [Fact]
+    public void TryToCreateSentinelFile_CreateSameSentinel_Fails()
+    {
+        var directory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+        Directory.CreateDirectory(directory);
+
+        var id = Guid.NewGuid().ToString();
+
+        var result1 = DebugLogReader.TryToCreateSentinelFile(directory, id);
```

Comment:
NIT: I'd go as far as starting both in a few threads (synchronized with a Barrier) then assert that a single one returned true.